moped-editor/src/views/projects/projectView/ProjectComponentEdit.js

```diff
@@ -392,55 +286,44 @@ const ProjectComponentEdit = ({
   /**
    * Calls upsert project features mutation, refetches data, and handles dialog close on success
    */
-  const generateMapUpserts = () => {
-    const editedFeatures = editFeatureCollection.features;
-    const featureIdPropertyName =
-      editedFeatures[0].properties.sourceLayer === "ATD_ADMIN.CTN" ||
-      editedFeatures[0].properties.sourceLayer === "drawnByUserLine"
-        ? "CTN_SEGMENT_ID"
-        : "INTERSECTIONID";
```

Comment:
Chia just added this logic to check for `featureIdPropertyName`. over the course of refactoring, i modified `generateMapUpserts` (now `generateFeatureUpserts` in such a way that we don't need to know or care about this id. instead we're relying on `feature_id` (the primary key/auto-increment of `moped_proj_features`) to do the diff between the edit session feature state and the db feature state. doing this significantly easier because of the components schema change.

Reply:
as currently written, we do this diffing/payload prepping after the user clicks the "save" button, but i think there's a better, future, state in which we continuously update the DB payload as the user interacts with the map/dropdowns by extending the dispatch pattern used for managing map state.